File: backend/src/main/java/t/lab/guide/service/PointCategoryService.java
package t.lab.guide.service;

import t.lab.guide.dto.admin.category.AdminCreateCategoryRequest;
import t.lab.guide.dto.admin.category.AdminPatchCategoryRequest;
import t.lab.guide.dto.point.category.CategoryItem;
import t.lab.guide.dto.point.category.CategoryListResponse;

public interface PointCategoryService {

    CategoryListResponse getAllCategories();

    CategoryItem createCategory(AdminCreateCategoryRequest request);

    CategoryItem patchCategory(Long id, AdminPatchCategoryRequest request);

    void deleteCategory(Long id);

    //TODO когда будут entity(model), добавить сервисные методы
}

File: backend/src/main/kotlin/t/lab/guide/service/PointCategoryService.kt
package t.lab.guide.service

import t.lab.guide.dto.admin.category.AdminCreateCategoryRequest
import t.lab.guide.dto.admin.category.AdminPatchCategoryRequest
import t.lab.guide.dto.point.category.CategoryItem
import t.lab.guide.dto.point.category.CategoryListResponse

interface PointCategoryService {
    fun getAllCategories(): CategoryListResponse

    fun createCategory(request: AdminCreateCategoryRequest): CategoryItem

    fun patchCategory(
        id: Long,
        request: AdminPatchCategoryRequest,
    ): CategoryItem

    fun deleteCategory(id: Long)

// TODO когда будут entity(model), добавить сервисные методы
}
